src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -128,15 +129,17 @@ struct SourceMessageBatch<Key, Value, Diff> {
     /// The current upper of the `SourceReader`, at the time this batch was
     /// emitted. Source uppers of emitted batches must never regress.
     source_upper: OffsetAntichain,
+    /// The largest offsets of messages in this batch, per partition.
+    max_offsets: OffsetAntichain,
```

Comment:
I think we want to put the frontier in the message it should be a true frontier with the `+1`s included so that we have `source_upper` and `batch_upper`.

The code before the parallelization was doing the `+1` internally because the `batch_upper` was not being provided by the caller, it was being calculated by iterating over the `messages`. https://github.com/MaterializeInc/materialize/blob/305082a6a99ee063839975c86bd1e821a2af0e23/src/storage/src/source/reclock.rs#L135-L151